krabcake/rs_hello/src/lib.rs

```diff
@@ -448,13 +449,44 @@ pub extern "C" fn rs_client_request_print_tag_of(
             vgPlain_umsg(
                 b"print_tag_of `%s` (0x%08llx), no tag found\n\0".as_ptr() as *const c_char,
                 name_addr,
-                address_of_interest,
+                STACKS.get_stack_dbg_id_or_assign(address_of_interest),
             );
             true
         }
     }
 }
 
+struct VgPlainUmsgWriter;
+
+impl fmt::Write for VgPlainUmsgWriter {
+    fn write_str(&mut self, s: &str) -> fmt::Result {
+        assert!(s.is_ascii());
+
+        const CHUNK_SIZE: usize = 128;
+        // Everything below should work for any positive value for CHUNK_SIZE
+        // (including 1). We do not actually expect to ever emit long messages
+        // to the user, so we don't need to spend much time figuring out ieal
+        // chunking sizes.
+
+        let mut buf: [u8; CHUNK_SIZE+1] = [b'\0'; CHUNK_SIZE+1];
+        let bytes = s.as_bytes();
+        for chunk in bytes.chunks(CHUNK_SIZE) {
+            // set aside destination slice in buf
+            let byte_count = chunk.len();
+            let (dest, follow) = buf.split_at_mut(byte_count);
+            // copy exactly the chunk into the destination
+            dest.copy_from_slice(chunk);
+            // make sure its null-terminated
+            follow[0] = b'\0';
+            // now print the chunk, with the copied bytes, out.
+            unsafe {
+                vgPlain_umsg("%s\0".as_ptr() as *const c_char, buf.as_ptr());
```

Comment:
It might be good to have a test (in the outer `kc` crate that has the test suite) that exercises what  is special about this line.

Namely, the reason we pass "%s\0" instead of just passing in `buf` directly to `vgPlain_usmg` is to handle the case where `buf` itself contains printf specifiers like `%s` or `%d` etc.

Reply:
This is https://github.com/pnkfelix/krabcake/pull/23